airflow/providers/amazon/aws/example_dags/example_sagemaker.py

```diff
@@ -0,0 +1,179 @@
+    "AppSpecification": {
+        "ImageUri": f"{image_uri}",
+        "ContainerEntrypoint": ["python3", "./preprocessing.py"],
+    },
+    "RoleArn": f"{role}",
+}
+
+sagemaker_training_job_config = {
+    "AlgorithmSpecification": {
+        "TrainingImage": f"{image_uri}",
+        "TrainingInputMode": "File",
+    },
+    "InputDataConfig": [
+        {
+            "ChannelName": "config",
+            "DataSource": {
+                "S3DataSource": {
+                    "S3DataType": "S3Prefix",
+                    "S3Uri": f"s3://{s3_bucket}/config/",
+                    "S3DataDistributionType": "FullyReplicated",
+                }
+            },
+            "CompressionType": "None",
+            "RecordWrapperType": "None",
+        },
+    ],
+    "OutputDataConfig": {
+        "KmsKeyId": "",
+        "S3OutputPath": f"s3://{s3_bucket}/training/",
+    },
+    "ResourceConfig": {
+        "InstanceType": "ml.m5.large",
+        "InstanceCount": 1,
+        "VolumeSizeInGB": 5,
+    },
+    "StoppingCondition": {"MaxRuntimeInSeconds": 6000},
+    "RoleArn": f"{role}",
+    "EnableNetworkIsolation": False,
+    "EnableInterContainerTrafficEncryption": False,
+    "EnableManagedSpotTraining": False,
+    "TrainingJobName": training_job_name,
+}
+
+sagemaker_create_model_config = {
+    "ModelName": model_name,
+    "Containers": [
+        {
+            "Image": f"{image_uri}",
+            "Mode": "SingleModel",
+            "ModelDataUrl": f"s3://{s3_bucket}/training/{training_job_name}/output/model.tar.gz",
+        }
+    ],
+    "ExecutionRoleArn": f"{role}",
+    "EnableNetworkIsolation": False,
+}
+
+sagemaker_inference_config = {
+    "TransformJobName": "sample_transform_job",
+    "ModelName": model_name,
+    "TransformInput": {
+        "DataSource": {
+            "S3DataSource": {
+                "S3DataType": "S3Prefix",
+                "S3Uri": f"s3://{s3_bucket}/config/config_date.yml",
+            }
+        },
+        "ContentType": "application/x-yaml",
+        "CompressionType": "None",
+        "SplitType": "None",
+    },
+    "TransformOutput": {"S3OutputPath": f"s3://{s3_bucket}/inferencing/output/"},
+    "TransformResources": {"InstanceType": "ml.m5.large", "InstanceCount": 1},
+}
+
+# [START howto_operator_sagemaker]
+with DAG(
+    "sample_sagemaker_dag",
+    schedule_interval=None,
+    start_date=datetime(2022, 2, 21),
+    catchup=False,
+) as dag:
+    sagemaker_processing_task = SageMakerProcessingOperator(
+        config=sagemaker_processing_job_config,
+        aws_conn_id="aws_default",
+        task_id="sagemaker_preprocessing_task",
+    )
+
+    training_task = SageMakerTrainingOperator(
+        config=sagemaker_training_job_config, aws_conn_id="aws_default", task_id="sagemaker_training_task"
+    )
+
+    model_delete_task = SageMakerDeleteModelOperator(
+        task_id="sagemaker_delete_model_task",
+        model_name=model_name,
```

Comment:
I don't follow on that.
Was `sample_model` existed before this DAG started?

the goal of example dags is that users should be able to run them and it will work for them.
Is this example really runnable?

Reply:
I got the intent. Thanks!

So with previous changes the delete model in case the model was not present does not use to through ValidationException or ResourceNotFound exception when the model was not present and was simply used to return false so the example DAG was working fine and I have tested it at my end with sample bucket and s3 path. But with the latest changes the exception will come so I have updated the example based on latest changes as well. Please check and resolve. Thanks :)